mypy/checkexpr.py

```diff
@@ -185,7 +185,7 @@ def visit_call_expr(self, e: CallExpr, allow_none_return: bool = False) -> Type:
         """Type check a call expression."""
         if e.analyzed:
             # It's really a special form that only looks like a call.
-            return self.accept(e.analyzed, self.type_context[-1])
+            return self.accept(e.analyzed, self.type_context[-1], always_allow_any=True)
```

Comment:
What, exactly, is this allowing?  At a minimum, I think this needs a comment about why `always_allow_any=True`.